scipy/sparse/linalg/_isolve/tests/test_iterative.py

```diff
@@ -621,6 +633,7 @@ def test_basic(self):
 
         assert_allclose(x_gm[0], 0.359, rtol=1e-2)
 
+    @pytest.mark.filterwarnings(f"ignore:{CB_TYPE_FILTER}:DeprecationWarning")
```

Comment:
This innocent-looking line breaks our `test_warnings` infrastructure; I've fixed it to the best of my (limited) understanding of AST processing, but adding @rgommers / @andyfaff for opinions/help in reviewing `scipy/_lib/tests/test_warnings.py`.